codyze-compliance/src/main/kotlin/de/fraunhofer/aisec/codyze/compliance/Command.kt

```diff
@@ -0,0 +1,113 @@
+ *                    \______/ \__|       \______/
+ *
+ */
+package de.fraunhofer.aisec.codyze.compliance
+
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.core.subcommands
+import com.github.ajalt.clikt.parameters.groups.provideDelegate
+import de.fraunhofer.aisec.codyze.*
+
+/** The main `compliance` command. */
+class ComplianceCommand : CliktCommand() {
+    override fun run() {}
+}
+
+/**
+ * A command that operates on a project. This class provides the common options and functions for
+ * all commands.
+ */
+abstract class ProjectCommand : CliktCommand() {
+    private val projectOptions by ProjectOptions()
+    private val translationOptions by TranslationOptions()
+
+    /** Loads the security goals from the project. */
+    fun loadSecurityGoals(): List<SecurityGoal> {
+        return loadSecurityGoals(projectOptions.directory.resolve("security-goals"))
+    }
+
+    /**
+     * This method is called by the `run` method to perform the actual analysis. It is separated to
+     * allow for easier access from overriding applications.
+     */
+    protected fun analyze(): AnalysisResult {
+        // Load the security goals from the project
+        val goals = loadSecurityGoals(projectOptions.directory.resolve("security-goals"))
+
+        // Analyze the project
+        val project = AnalysisProject.fromOptions(projectOptions, translationOptions)
+        val result = project.analyze()
+        val tr = result.translationResult
+
+        // Connect the security goals to the translation result for now. Later we will add them to
+        // individual concepts
+        for (goal in goals) {
+            goal.underlyingNode = tr
+
+            // Load and execute queries associated to the goals
+            for (objective in goal.objectives) {
+                objective.underlyingNode = tr
+
+                val scriptFile =
+                    projectOptions.directory
+                        .resolve("queries")
+                        .resolve(
+                            "${objective.name.localName.lowercase().replace(" ", "-")}.query.kts"
+                        )
+                for (stmt in objective.statements.withIndex()) {
+                    tr.evalQuery(scriptFile.toFile(), "statement${stmt.index + 1}")
+                }
+            }
+        }
+
+        return result
+    }
+}
+
+/** The `scan` command. This will scan the project for compliance violations in the future. */
+open class ScanCommand : ProjectCommand() {
+    override fun run() {
+        val result = analyze()
+
+        result.run.results?.forEach { echo(it.message) }
+    }
+}
+
+/**
+ * The `list-security-goals` command. This will list the names of all security goals in the
+ * specified project.
+ *
+ * This command assumes that the project contains a folder named `security-goals` that contains YAML
+ * files with the security goals.
+ */
+class ListSecurityGoals : ProjectCommand() {
```

Comment:
I wouldn't inherit from `ProjectCommand` because I currently see no reason to specify a project dir or CPG translation options. We're also not calling `analyze()`.

Should we use use a `CliktCommand` for now?

Reply:
We are calling `loadSecurityGoals` which loads the security goals in the project specified by the project options, so we are using the project dir.

Reply:
Missed that one 😔 

It might still be confusing to have the CPG Translation Options in the subcommand for `list-security-goals`.

Reply:
They don't hurt at the moment and I don't want to introduce too many different commands.